Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm

```diff
@@ -1519,7 +1519,23 @@ - (void)URLSession:(NSURLSession *)session webSocketTask:(NSURLSessionWebSocketT
 #endif
 }
 
-NetworkSessionCocoa::~NetworkSessionCocoa() = default;
+NetworkSessionCocoa::~NetworkSessionCocoa()
+{
+    notifyAdAttributionKitOfSessionTermination();
```

Comment:
Is there any risk we get killed before we can complete this notification? I'm thinking of when we get jetsammed. Just trying to understand if this will be "leaky" or not.

For crashes, I guess all bets are off.

Reply:
That is a problem that is not solved yet, but the common case of closing the browser hits this.